### PR TITLE
feat(manager/pip-compile): enable support for setup.py

### PR DESCRIPTION
**Pull request by not7cd**
## Changes

Support for `setup.py` has been enabled. 

## Context

This was disabled in #26858

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [x] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/modules/manager/pip-compile/extract.spec.ts [hunk 1]
```diff
@@ -27,13 +27,22 @@ describe('modules/manager/pip-compile/extract', () => {
...
     });
 
+    it('returns object for setup.py', () => {
+      const packageFile = extractPackageFile(
+        Fixtures.get('setup.py'),
+        'lib/setup.py',
+        {},
+      );
+      expect(packageFile).toHaveProperty('deps');
+      expect(packageFile?.deps).toMatchSnapshot();
```
Comment (viceice) [inline lib/modules/manager/pip-compile/extract.spec.ts:37]: no snapshots

File: lib/modules/manager/pip-compile/__fixtures__/setup.py [hunk 2]
```diff
@@ -0,0 +1,108 @@
+
```
Comment (viceice) [inline lib/modules/manager/pip-compile/__fixtures__/setup.py:1]: please remove everything we don't need to reduce size

File: lib/modules/manager/pip-compile/__fixtures__/setup.py
Comment (viceice) [inline lib/modules/manager/pip-compile/__fixtures__/setup.py:1]: Please also update dep versions so we don't get anoying warnings for them

- https://github.com/renovatebot/renovate/actions/runs/7895303867/job/21547465888?pr=27282

File: lib/modules/manager/pip-compile/extract.spec.ts [hunk 3]
```diff
@@ -27,13 +27,22 @@ describe('modules/manager/pip-compile/extract', () => {
       expect(packageFile?.deps[0]).toHaveProperty('depName', 'attrs');
     });
 
+    it('returns object for setup.py', () => {
+      const packageFile = extractPackageFile(
+        Fixtures.get('setup.py'),
```
Comment (viceice) [inline lib/modules/manager/pip-compile/extract.spec.ts:32]:
```suggestion
        Fixtures.get('setup.py', '../pip_setup'),
```

read test fixture from `pip_setup` manager 😉

File: lib/modules/manager/pip-compile/extract.ts [hunk 4]
```diff
@@ -1,8 +1,8 @@
 import { logger } from '../../../logger';
 import { readLocalFile } from '../../../util/fs';
 import { extractPackageFile as extractRequirementsFile } from '../pip_requirements/extract';
-// TODO(not7cd): enable in the next PR, when this can be properly tested
-// import { extractPackageFile as extractSetupPyFile } from '../pip_setup';
+import { extractPackageFile as extractSetupPyFile } from '../pip_setup';
+// TODO: enable when this can be properly tested
 // import { extractPackageFile as extractSetupCfgFile } from '../setup-cfg';
```
Comment (viceice) [inline lib/modules/manager/pip-compile/extract.ts:6]: if you don't plan to implement this, then remove those comments

File: lib/modules/manager/pip-compile/extract.ts [hunk 5]
```diff
@@ -36,11 +36,10 @@ export function extractPackageFile(
...
-  // TODO(not7cd): extract based on manager: pep621, setuptools, identify other missing source types
+  // TODO: extract based on manager: pep621, setuptools cfg
   switch (manager) {
-    // TODO(not7cd): enable in the next PR, when this can be properly tested
-    // case 'pip_setup':
-    //   return extractSetupPyFile(content, _packageFile, _config);
+    case 'pip_setup':
+      return extractSetupPyFile(content, packageFile, _config);
     // case 'setup-cfg':
     //   return await extractSetupCfgFile(content);
```
Comment (viceice) [inline lib/modules/manager/pip-compile/extract.ts:44]: see other comment

File: lib/modules/manager/pip-compile/readme.md [hunk 6]
```diff
@@ -24,7 +24,7 @@ If Renovate matches a `pip-compile` output file it will extract original command
 - Use default header generation, don't use `--no-header` option.
 - Pass all source files explicitly.
 
-In turn `pip-compile` manager will find all source files and parse them as package files. Currently only `*.in` files associated with `pip_requirements` manager are handled.
+In turn `pip-compile` manager will find all source files and parse them as package files. Currently only `*.in` files associated with `pip_requirements` manager and `setup.py` files are handled.
```
Comment (viceice) [inline lib/modules/manager/pip-compile/readme.md:27]:
```suggestion
In turn `pip-compile` manager will find all source files and parse them as package files.
Currently only `*.in` files associated with `pip_requirements` manager and `setup.py` files associated with `pip_setup` manager are handled.
```

**Review by viceice — APPROVED** [review]

**Review by secustor — APPROVED** [review]

**Pull request merged**